### PR TITLE
Adding a rule to detect if an AWS Transfer Server is public

### PR DESCRIPTION
**Pull request by ac-square**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

https://aws.amazon.com/premiumsupport/knowledge-center/aws-sftp-endpoint-type/

### CODE REVIEW

File: tests/terraform/checks/resource/aws/test_TransferServerIsPublic.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import unittest
+
+from checkov.common.models.enums import CheckResult
+from checkov.terraform.checks.resource.aws.TransferServerIsPublic import check
+import hcl2
+
+
+class TestTransferServerIsPublic(unittest.TestCase):
```
Comment (gruebel) [inline tests/terraform/checks/resource/aws/test_TransferServerIsPublic.py:8]: Please change the tests to use the newer set up like here #1182

And also please add a test without specifying the `endpoint_type` should result in a failure.

**Comment by ac-square** [comment]
@gruebel updated per your comments

**Comment by gruebel** [comment]
nice, looks good 👍

**Comment by schosterbarak** [comment]
Thanks guys!

**Pull request merged**